src/python/pants/goal/run_tracker.py

```diff
@@ -304,8 +304,9 @@ def store_stats(self):
     # Upload to remote stats db.
     stats_url = self.get_options().stats_upload_url
     if stats_url:
-      t = threading.Thread(target=self.post_stats, args=(stats_url, stats, self.get_options().stats_upload_timeout))
-      t.start()
+      pid = os.fork()
+      if pid == 0:
+        self.post_stats(stats_url, stats, timeout=self.get_options().stats_upload_timeout)
```

Comment:
We'd probably want to do some kind of proper exit here, since the child would continue to execute after the fork. You could do something like the async clean-all patch did https://github.com/pantsbuild/pants/commit/c06ac5dde91583fb5bb7ae036496a1d58ffaf8a7#diff-214714c04a4aa1e4879b7773ba163ae6R47